src/derivatives.h

```diff
@@ -99,28 +98,17 @@ struct derivatives_nonspatial: public lower_chol_nonspatial<Type>, virtual deriv
     matrix<Type> sigma_d2 = allret["derivative2"];
     this->sigmad1_cache[this->full_visit] = sigma_d1;
     this->sigmad2_cache[this->full_visit] = sigma_d2;
-    this->sel_mat_cache[this->full_visit] = get_select_matrix<Type>(this->full_visit, this->n_visits);
-  }
-  // Cache and return the select matrix.
-  Eigen::SparseMatrix<Type> get_sel_mat(std::vector<int> visits) {
-    if (this->sel_mat_cache.count(visits) > 0) {
-      return this->sel_mat_cache[visits];
-    } else {
-      this->sel_mat_cache[visits] = get_select_matrix<Type>(visits, this->n_visits);
-      return this->sel_mat_cache[visits];
-    }
   }
   // Cache and return the first order derivatives using select matrix.
   matrix<Type> get_sigma_derivative1(std::vector<int> visits, matrix<Type> dist) override {
     auto target = this->sigmad1_cache.find(visits);
      if (target != this->sigmad1_cache.end()) {
       return target->second;
     } else {
-      Eigen::SparseMatrix<Type> sel_mat = this->get_sel_mat(visits);
       int n_visists_i = visits.size();
```

Comment:
there seems to be a typo here
```suggestion
      int n_visits_i = visits.size();
```
and similarly rename below